src/time.rs

```diff
@@ -0,0 +1,58 @@
+//! Time-Related Functions and Types
+
+pub use chrono::{DateTime, NaiveDateTime, TimeZone, Utc};
+pub use prost_types::Timestamp;
+
+/// Converts a prost [`Timestamp`] value to a chrono [`DateTime<Utc>`] value.
+/// Will return [`None`] if [`Timestamp`] `nanos` value is negative
+pub fn timestamp_to_datetime(ts: &Timestamp) -> Option<DateTime<Utc>> {
+    let nanos: u32 = ts.nanos.try_into().ok()?;
+    let ndt: NaiveDateTime = NaiveDateTime::from_timestamp_opt(ts.seconds, nanos)?;
```

Comment:
Using '?' will return the non `Ok`/ `Some` variant, which in this case is `None` as `from_timestamp_opt` is returning an `Option` as well.
The caller can check the function documentation, which states it will return `None` if `nanos` value is negative